openmetadata-ui/src/main/resources/ui/src/components/Alerts/AlertsDetails/AlertDetails.test.tsx

```diff
@@ -0,0 +1,139 @@
+  getDisplayNameForEntities: jest.fn().mockImplementation((entity) => entity),
+  getFunctionDisplayName: jest
+    .fn()
+    .mockImplementation((filterFQN) => filterFQN),
+}));
+
+jest.mock('../../common/TitleBreadcrumb/TitleBreadcrumb.component', () =>
+  jest.fn().mockReturnValue(<span>TitleBreadcrumb</span>)
+);
+
+jest.mock('../../PageHeader/PageHeader.component', () =>
+  jest.fn().mockReturnValue(<span>PageHeader</span>)
+);
+
+const mockDelete = jest.fn();
+const mockPush = jest.fn();
+
+const mockProps1 = {
+  alerts: {
+    id: 'alertId',
+    name: 'alertName',
+    alertType: AlertType.ActivityFeed,
+    destinations: [] as Destination[],
+    filteringRules: {
+      resources: ['resource1', 'resource2', 'resource3'],
+      rules: [
+        {
+          fullyQualifiedName: 'conditionName',
+          condition: 'condition',
+          effect: Effect.Include,
+          name: 'ruleName',
+        },
+      ],
+    },
+  },
+  onDelete: mockDelete,
+};
+
+const mockProps2 = {
+  ...mockProps1,
+  pageHeaderData: {} as HeaderProps['data'],
+  allowDelete: false,
+  breadcrumb: {} as TitleBreadcrumbProps['titleLinks'],
+  allowEdit: false,
+};
+
+describe('AlertDetailsComponent', () => {
+  it('should render all necessary elements based on mockProps1', () => {
+    render(<AlertDetailsComponent {...mockProps1} />, {
+      wrapper: MemoryRouter,
+    });
+
+    expect(screen.queryByText('TitleBreadcrumb')).not.toBeInTheDocument();
+    expect(screen.queryByText('PageHeader')).not.toBeInTheDocument();
+
+    const editButton = screen.getByRole('button', { name: 'label.edit' });
+    const deleteButton = screen.getByRole('button', { name: 'label.delete' });
+
+    expect(editButton).toBeInTheDocument();
+    expect(deleteButton).toBeInTheDocument();
+
+    expect(screen.getByText('label.trigger')).toBeInTheDocument();
+    expect(
+      screen.getByText('resource1, resource2, resource3')
+    ).toBeInTheDocument();
+
+    expect(screen.getByText('label.filter-plural')).toBeInTheDocument();
+
+    expect(screen.getByText('conditionName === condition')).toBeInTheDocument();
+
+    expect(screen.getByText('label.destination')).toBeInTheDocument();
+  });
+
+  it('should render breadcrumb and pageHeader if data passed in props', () => {
+    render(<AlertDetailsComponent {...mockProps2} />, {
+      wrapper: MemoryRouter,
+    });
+
+    expect(screen.getByText('TitleBreadcrumb')).toBeInTheDocument();
+    expect(screen.getByText('PageHeader')).toBeInTheDocument();
+
+    const editButton = screen.queryByRole('button', { name: 'label.edit' });
+    const deleteButton = screen.queryByRole('button', { name: 'label.delete' });
+
+    expect(editButton).not.toBeInTheDocument();
+    expect(deleteButton).not.toBeInTheDocument();
+  });
+
+  it('should call action handlers on click of buttons', () => {
+    const history = createMemoryHistory();
+
+    history.push = mockPush;
+
+    render(
+      <Router history={history}>
```

Comment:
Do we need this? I think we can use memoryRouter here.

Reply:
We want to check here Link is redirecting onclick or not, MemoryRouter not accept history prop.

Reply:
Then in that case can we mock the `useHistory` hook push method?

Reply:
We are not using userHistory hook in this component, I am testing there <Link> redirect path that will be create based on props passed in AlertDetails component.

To verify it, path creating currectly, clicking on link and checking if push method of Router getting called with the value of "to" prop in Link.

Reply:
Okay, got it, thanks for the explanation @Abhishek332